flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/sink/TableStoreSink.java

```diff
@@ -0,0 +1,175 @@
+    }
+
+    @Override
+    public ChangelogMode getChangelogMode(ChangelogMode requestedMode) {
+        if (!tableStore.valueCountMode()
+                && logChangelogMode == LogOptions.LogChangelogMode.UPSERT) {
+            ChangelogMode.Builder builder = ChangelogMode.newBuilder();
+            for (RowKind kind : requestedMode.getContainedKinds()) {
+                if (kind != RowKind.UPDATE_BEFORE) {
+                    builder.addContainedKind(kind);
+                }
+            }
+            return builder.build();
+        }
+        return requestedMode;
+    }
+
+    @Override
+    public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
+        LogSinkProvider logSinkProvider = null;
+        if (logStoreTableFactory != null) {
+            logSinkProvider =
+                    logStoreTableFactory.createSinkProvider(
+                            logStoreContext,
+                            new LogStoreTableFactory.SinkContext() {
+                                @Override
+                                public boolean isBounded() {
+                                    return !streaming;
+                                }
+
+                                @Override
+                                public <T> TypeInformation<T> createTypeInformation(
+                                        DataType consumedDataType) {
+                                    return context.createTypeInformation(consumedDataType);
+                                }
+
+                                @Override
+                                public <T> TypeInformation<T> createTypeInformation(
+                                        LogicalType consumedLogicalType) {
+                                    return context.createTypeInformation(consumedLogicalType);
+                                }
+
+                                @Override
+                                public DynamicTableSink.DataStructureConverter
+                                        createDataStructureConverter(DataType consumedDataType) {
+                                    return context.createDataStructureConverter(consumedDataType);
+                                }
+                            });
+        }
+        final LogSinkProvider finalLogSinkProvider = logSinkProvider;
+        return (DataStreamSinkProvider)
+                (providerContext, dataStream) ->
+                        tableStore
+                                .sinkBuilder()
+                                .withInput(
+                                        new DataStream<>(
+                                                dataStream.getExecutionEnvironment(),
+                                                dataStream.getTransformation()))
+                                .withLockFactory(lockFactory)
+                                .withLogSinkProvider(finalLogSinkProvider)
+                                .withOverwritePartition(staticPartitions)
```

Comment:
Empty `staticPartitions` is overwrite all... 
Here should be `overwrite ? staticPartitions : null`
Add a TODO in test class to add cases.